examples/glue/run_glue_trainer.py

```diff
@@ -185,41 +185,62 @@ def merge_hparams(hparams: TrainerHparams, override_hparams: GLUETrainerHparams)
 
 def spawn_finetuning_jobs(
     task_to_save_ckpt: Dict[str, bool],
-    ckpt_load_path: str,
+    ckpt_load_paths: List[str],
     ckpt_save_folder: str,
     glue_metrics: GlueState,
     base_yaml_file: str,
     save_locally: bool,
-    parent_ckpt: Optional[str] = None,
+    parent_ckpts: Optional[List[str]] = None,
     load_ignore_keys: Optional[List[str]] = None,
 ) -> None:
-    """Set up CUDA environment and process pool for given finetuning jobs."""
+    """Set up CUDA environment and process pool for given finetuning jobs and wait for them to complete."""
+    if parent_ckpts:
+        assert len(parent_ckpts) == len(ckpt_load_paths), 'Must supply one parent_ckpt per ckpt_load_path'
+    else:
+        parent_ckpts = ckpt_load_paths  # By default, the "parent checkpoint" is logged simply as the checkpoint
+
+    # To reduce noisiness with these tasks, expand the evaluation to multiple fine-tuning seeds per pre-train checkpoint
+    seed_overrides = {
+        'sst-2': [19, 8364, 717],
+        'cola': [19, 8364, 717, 10536],
+        'rte': [19, 8364, 717, 10536, 90166],
+        'mrpc': [19, 8364, 717, 10536, 90166],
+        'stsb': [19, 8364, 717, 10536, 90166],
+    }
+
+    # Set up CUDA environment(s) and process pool
     ctx = mp.get_context('spawn')
     cuda_envs = init_cuda_queue(torch.cuda.device_count(), ctx)
-    finetune_tasks = list(task_to_save_ckpt.keys())
-    num_tasks = len(finetune_tasks)
-
-    if parent_ckpt:
-        wandb_group_name = parent_ckpt
-        logged_ckpt_name = parent_ckpt
-    else:
-        wandb_group_name = ckpt_load_path
-        logged_ckpt_name = ckpt_load_path
-
-    # finetuning from pretrained checkpoint
-    print(f'FINETUNING ON {ckpt_load_path}!')
-    done_callback = lambda future: log_metrics(
-        metric=future.result(), ckpt_filename=logged_ckpt_name, glue_metrics=glue_metrics)
     free_port = get_free_tcp_port()
     executor = Pool(max_workers=torch.cuda.device_count(),
                     initializer=init_cuda_env,
                     initargs=(cuda_envs, free_port),
                     mp_context=ctx)
-    for rank in range(num_tasks):
-        task = finetune_tasks[rank]
-        future = executor.submit(train_finetune, base_yaml_file, task, task_to_save_ckpt[task], ckpt_load_path,
-                                 wandb_group_name, ckpt_save_folder, save_locally, free_port + rank, load_ignore_keys)
-        future.add_done_callback(done_callback)
+
+    def make_callback(parent_ckpt, glue_metrics):
+        return lambda future: log_metrics(future.result(), ckpt_filename=parent_ckpt, glue_metrics=glue_metrics)
+
+    # Fine-tune from pre-trained checkpoint(s)
+    for parent_idx, (ckpt_load_path, parent_ckpt) in enumerate(zip(ckpt_load_paths, parent_ckpts)):
```

Comment:
Can we move the `zip` and `(ckpt_load_path, parent_ckpt)` into their own line? This line has a lot of work going on.